k8s/secrets.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2019 OpenFaaS Authors
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package k8s
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	types "github.com/openfaas/faas-provider/types"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	secretsMountPath = "/var/openfaas/secrets"
+	secretLabel      = "app.kubernetes.io/managed-by"
+	secretLabelValue = "openfaas"
+)
+
+// SecretsClient exposes the standardized CRUD behaviors for Kubernetes secrets.  These methods
+// will ensure that the secrets are structured and labelled correctly for use by the OpenFaaS system.
+type SecretsClient interface {
+	// List returns a list of available function secrets.  Only the names are returned
+	// to ensure we do not accidentally read or print the sensitive values during
+	// read operations.
+	List(namespace string) (names []string, err error)
+	// Create adds a new secret, with the appropriate labels and structure to be
+	// used as a function secret.
+	Create(secret types.Secret) error
+	// Replace updates the value of a function secret
+	Replace(secret types.Secret) error
+	// Delete removes a function secret
+	Delete(name string, namespace string) error
```

Comment:
@alexellis for symmetry with the other methods, what do you of passing `types.Secret` here instead of `name`, `namespace`?  In some ways it will be cleaner because then the caller will have named fields to set, which lowers the likelyhood of swapping the namespace and the name. For example `Delete("foo", "openfaas-fn")` and `Delete("openfaas-fn", "foo")` are both valid and the error is hard to detect